packages/snaps-sdk/src/types/device.ts

```diff
@@ -0,0 +1,97 @@
+import type { Struct } from '@metamask/superstruct';
+import { literal, refine, string, union } from '@metamask/superstruct';
+
+import type { Describe } from '../internals';
+
+/**
+ * The type of the device.
+ */
+export type DeviceType = 'hid' | 'bluetooth';
+
+/**
+ * A struct that represents the `DeviceType` type.
+ */
+export const DeviceTypeStruct: Describe<DeviceType> = union([
+  literal('hid'),
+  literal('bluetooth'),
+]);
+
+/**
+ * The ID of the device. It consists of the type of the device, the vendor ID,
+ * and the product ID.
+ */
+export type DeviceId = `${DeviceType}:${string}:${string}`;
+
+/**
+ * The ID of the device that is scoped to the type of the device.
+ *
+ * @example
+ * type HidDeviceId = ScopedDeviceId<'hid'>;
+ * // => `hid:${string}:${string}`
+ */
+export type ScopedDeviceId<Type extends DeviceType> =
+  `${Type}:${string}:${string}` extends DeviceId
+    ? `${Type}:${string}:${string}`
+    : never;
+
+/**
+ * A struct that represents the `DeviceId` type.
+ *
+ * @param type - The type of the device.
+ * @returns A struct that represents the `DeviceId` type.
+ */
+export function deviceId<Type extends DeviceType>(
+  type?: Type,
+): Type extends DeviceType ? Struct<ScopedDeviceId<Type>> : Struct<DeviceId> {
+  return refine(string(), 'device ID', (value) => {
+    if (type) {
```

Comment:
I'm not sure I follow what this struct is meant to validate

Reply:
Right now it just validates that a string consists of three parts (`1:2:3`).